virtcontainers/netmon/netmon.go

```diff
@@ -0,0 +1,530 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package main
+
+import (
+	"encoding/json"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"net"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/kata-containers/agent/protocols/grpc"
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+const (
+	netmonName    = "kata-netmon"
+	netmonVersion = "0.0.1"
+
+	kataCLIAddIfaceCmd   = "add-interface"
+	kataCLIDelIfaceCmd   = "del-interface"
+	kataCLIUpdtRoutesCmd = "update-routes"
+
+	kataSuffix = "kata"
+
+	netlinkFamily = netlink.FAMILY_V4
+
+	storageParentPath = "/var/run/kata-containers/netmon/sbs"
+	storageDirPerm    = 0750
+	sharedFile        = "shared.json"
+	storageFilePerm   = 0640
+)
+
+type netmonParams struct {
+	sandboxID   string
+	runtimePath string
+	netnsPath   string
+	debug       bool
+}
+
+type netmon struct {
+	netmonParams
+
+	storagePath string
+	sharedFile  string
+
+	netIfaces map[int]grpc.Interface
+
+	linkUpdateCh chan netlink.LinkUpdate
+	linkDoneCh   chan struct{}
+
+	rtUpdateCh chan netlink.RouteUpdate
+	rtDoneCh   chan struct{}
+
+	netHandler *netlink.Handle
+}
+
+func printVersion() {
+	fmt.Printf("%s version %s\n", netmonName, netmonVersion)
+}
+
+func parseOptions() netmonParams {
+	var version bool
+
+	params := netmonParams{}
+
+	flag.BoolVar(&params.debug, "d", false, "enable debug mode")
+	flag.BoolVar(&version, "v", false, "display program version and exit")
+	flag.StringVar(&params.sandboxID, "s", "", "sandbox id (required)")
+	flag.StringVar(&params.runtimePath, "r", "", "runtime path (required)")
+	flag.StringVar(&params.netnsPath, "n", "", "network namespace path (required)")
+
+	flag.Parse()
+
+	if version {
+		printVersion()
+		os.Exit(0)
+	}
+
+	if params.sandboxID == "" {
+		fmt.Fprintf(os.Stderr, "Error: sandbox id is empty, one must be provided\n")
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	if params.runtimePath == "" {
+		fmt.Fprintf(os.Stderr, "Error: runtime path is empty, one must be provided\n")
+		flag.PrintDefaults()
+		os.Exit(1)
+	}
+
+	/*
```

Comment:
why is this commented? is this a TODO ?

Reply:
Oh yeah, I forgot to remove it. I'll do that when I'll update the PR.

Reply:
done